### PR TITLE
Hw3

### PR DESCRIPTION
**Pull request by gleason9113**
HW3

01-routing.js
-Built server routes for the specified resources.

01-routing2.js
-Rebuilt file using Express (for funsies). This is not part of the assignment; feel free to ignore this file.

02-url.js
-Wrote switch statement to parse url query parameters and format results into table.

03-form.js
-Wrote server to redirect / to /form, get form data and post to /submit.  Yes, I did write this using Express and body-parser; I ran out of time to figure out how to do it in vanilla Node, and will have to take the points hit, if need be.

Feedback requested:  Technical and style critiques- as always, if there's a better way to do something, let me know!

### CODE REVIEW

File: hw3/01-routing.js [hunk 1]
```diff
@@ -34,13 +22,62 @@ const server = http.createServer((req, res) => {
...
+      res.write(
+        `Unless you clicked the redirected link on the main page. Then you just travelled here.`
+      );
+      res.end();
+      break;
+    case "/cache":
+      res.writeHead(
+        200,
+        { "Content-Type": "text/html" },
+        { "Cache-Control": "max-age: 86_400" }
```
Comment (smah29) [inline hw3/01-routing.js:55]: The underscore in 86_400 is wrong.

File: hw3/03-form.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+const http = require("http");
+const express = require("express");
```
Comment (smah29) [inline hw3/03-form.js:2]: I saw in the slack channel that, third party modules like express should not be used in HW3

**Review by smah29 — COMMENTED** [review]
Given some review comments, otherwise well written code.

File: hw3/03-form.js (hunk 2 above)
Comment (gleason9113) [reply hw3/03-form.js:2]: There's a Slack channel for this class?  What is it?

Comment (gleason9113) [reply hw3/03-form.js]: Rewrote 03-form.js without Express.

File: hw3/01-routing.js (hunk 1 above)
Comment (gleason9113) [reply hw3/01-routing.js:55]: Not from a syntax standpoint.  Might be from a style standpoint, but it makes numbers so much more readable that I'm willing to take the hit.

**Comment by JeremyIV** [comment]
Ex1: Looks great, nice job!
Ex2: The intention with this exercise was that your web server be able to respond not just to these three example URLs, but to any parameter values passed in. e.g. if I change "spongebob=squarepants" to"eugene=krabbs", the table should update accordingly, not 404. 
Ex3:  When retrieving data from a form, you shouldn't have to parse that data yourself. You can use the 'querystring' module, and this will save you a lot of trouble, e.g. "bob@hotmail.com" appears as "bob%40hotmail.com" due to improper, home-brewed parsing. Otherwise, nice job.

**Pull request merged**